File: src/main/java/mediathek/gui/actions/ShowAboutAction.java
package mediathek.gui.actions;

import mediathek.mainwindow.MediathekGui;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class ShowAboutAction extends AbstractAction {
    private final MediathekGui mediathekGui;

    public ShowAboutAction(MediathekGui mediathekGui) {
        this.mediathekGui = mediathekGui;
        putValue(Action.NAME,"Über dieses Programm...");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        mediathekGui.showAboutDialog();
    }
}

File: src/main/kotlin/mediathek/gui/actions/ShowAboutAction.kt
package mediathek.gui.actions

import mediathek.gui.dialog.AboutDialog
import mediathek.swing.centerOnScreen
import java.awt.event.ActionEvent
import javax.swing.AbstractAction
import javax.swing.JFrame

class ShowAboutAction(
    private val parent: JFrame,
) : AbstractAction() {
    override fun actionPerformed(e: ActionEvent?) {
        val dialog = AboutDialog(parent)
        dialog.centerOnScreen()
        dialog.isVisible = true
        dialog.dispose()
    }

    init {
        putValue(NAME, "Über dieses Programm...")
    }
}
